addons/delivery/models/sale_order.py

```diff
@@ -48,7 +48,8 @@ def set_delivery_line(self, carrier, amount):
         self._remove_delivery_line()
 
         for order in self:
-            order._create_delivery_line(carrier, amount, price_unit_in_description=self.carrier_id.invoice_policy == 'real')
+            order.carrier_id = carrier.id
```

Comment:
you can remove this commit, it has been merged in saas-12.3 and just wait forwardport: https://github.com/odoo/odoo/pull/35855

Reply:
Done.